### DISCUSSION TITLE
Does PrivateGPT support finding and displaying ALL instances of items in a query?

### DISCUSSION THREAD

**Discussion by aronweiler** [imartinez/privateGPT#688] [Q&A]
I know the title may be a little unclear, but is PrivateGPT (or any LLM that operates using an embedding with a top_k) capable of searching through a list of loaded documents and returning a summary or a list of all instances of an item in a query?

For instance, if I load up 30 documents containing meeting minutes, and I want to know how many times a specific person spoke, would that be supported?

My query would look something like, "Please tell me how many times Person A proposed a rule change, and summarize the rule changes they asked for."

Any help is appreciated, thanks!

**Comment by sime2408** [comment]
You will need to build [a custom Tool for that](https://python.langchain.com/en/latest/modules/agents/tools/custom_tools.html) using langchain
